gems/aws-sigv4/lib/aws-sigv4/signer.rb

```diff
@@ -558,7 +558,9 @@ def host(uri)
 
       def standard_port?(uri)
         (uri.scheme == 'http' && uri.port == 80) ||
-        (uri.scheme == 'https' && uri.port == 443)
+        (uri.scheme == 'https' && uri.port == 443) ||
+        (uri.scheme == 'ws' && (uri.port == 80 || uri.port.nil?)) ||
+        (uri.scheme == 'wss' && (uri.port == 443 || uri.port.nil?))
       end
```

Comment:
After some discussion with @alextwoods, we think that it would be better to remove the `standard_port?` method.

The `host` method could be rewritten:

```
      def host(uri)
        if uri.default_port == uri.port # no longer calls standard_port?
          uri.host
        else
          "#{uri.host}:#{uri.port}"
        end
      end
```
This checks the URI for the default port and the port from the URI. If it's default, send the host, otherwise, include the port. Since `nil == nil => true` it would work for new schemes.

Pros:
 * In the future when other schemes are added, no updates are necessary.
 * Uses the `Generic#default_port` [method](https://github.com/ruby/uri/blob/master/lib/uri/generic.rb#L40)

Cons:
 * Assumes other schemes will inherit from `Generic` though this is probably a safe assumption.

What do you think?

Reply:
I like that approach way better. My initial assumption was to do something similar in the host method and skip the standard_port method all together, but then figured the pull request might be more likely to get accepted by appending to the existing approach. I'm all for this take instead.

Reply:
Better (backwards compatible) code is more likely to get accepted than hacky code :)